From 6110861065742136fce50ba4d137db72f38db422 Mon Sep 17 00:00:00 2001
From: Dmitriy Paliy <dmitriy.paliy@nokia.com>
Date: Fri, 1 Apr 2011 15:05:54 +0300
Subject: [PATCH] Add ERROR code response to ATD in maemo6 telephony

ERROR code response is added to ATD command in maemo6 telephony driver
based on HFP 1.5 specification, page 68:
"Standard error indication code. It shall be issued on detection of any
syntax, format or procedure error condition."
---
 audio/telephony-maemo6.c |   12 ++++--------
 1 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/audio/telephony-maemo6.c b/audio/telephony-maemo6.c
index 0cef7dd..be2fb87 100644
--- a/audio/telephony-maemo6.c
+++ b/audio/telephony-maemo6.c
@@ -618,7 +618,7 @@ static void remove_pending(DBusPendingCall *call)
 	pending_req_finalize(req);
 }
 
-static void last_number_call_reply(DBusPendingCall *call, void *user_data)
+static void create_call_reply(DBusPendingCall *call, void *user_data)
 {
 	DBusError err;
 	DBusMessage *reply;
@@ -648,7 +648,7 @@ void telephony_last_dialed_number_req(void *telephony_device)
 
 	ret = send_method_call(CSD_CALL_BUS_NAME, CSD_CALL_PATH,
 				CSD_CALL_INTERFACE, "CreateFromLast",
-				last_number_call_reply, telephony_device,
+				create_call_reply, telephony_device,
 				DBUS_TYPE_INVALID);
 	if (ret < 0)
 		telephony_dial_number_rsp(telephony_device,
@@ -687,16 +687,12 @@ void telephony_dial_number_req(void *telephony_device, const char *number)
 
 	ret = send_method_call(CSD_CALL_BUS_NAME, CSD_CALL_PATH,
 				CSD_CALL_INTERFACE, "Create",
-				NULL, NULL,
+				create_call_reply, telephony_device,
 				DBUS_TYPE_STRING, &number,
 				DBUS_TYPE_INVALID);
-	if (ret < 0) {
+	if (ret < 0)
 		telephony_dial_number_rsp(telephony_device,
 						CME_ERROR_AG_FAILURE);
-		return;
-	}
-
-	telephony_dial_number_rsp(telephony_device, CME_ERROR_NONE);
 }
 
 void telephony_transmit_dtmf_req(void *telephony_device, char tone)
-- 
1.7.1

